packages/wrangler/src/__tests__/publish.test.ts

```diff
@@ -3720,56 +3720,123 @@ export default{
-    // see https://github.com/cloudflare/wrangler2/issues/655 for bug details
+    it("should not match regular module specifiers when there aren't any possible legacy module matches", async () => {
+      // see https://github.com/cloudflare/wrangler2/issues/655 for bug details
 
-    fs.writeFileSync(
-      "./index.js",
-      `import inner from './inner/index.js'; export default {};`
-    );
-    fs.mkdirSync("./inner", { recursive: true });
-    fs.writeFileSync("./inner/index.js", `export default 123`);
-    mockSubDomainRequest();
-    mockUploadWorkerRequest();
+      fs.writeFileSync(
+        "./index.js",
+        `import inner from './inner/index.js'; export default {};`
+      );
+      fs.mkdirSync("./inner", { recursive: true });
+      fs.writeFileSync("./inner/index.js", `export default 123`);
+      mockSubDomainRequest();
+      mockUploadWorkerRequest();
 
-    await runWrangler(
-      "publish index.js --compatibility-date 2022-03-17 --name test-name"
-    );
-    expect(std.out).toMatchInlineSnapshot(`
-      "Uploaded test-name (TIMINGS)
-      Published test-name (TIMINGS)
-        test-name.test-sub-domain.workers.dev"
-    `);
-    expect(std.err).toMatchInlineSnapshot(`""`);
-    expect(std.warn).toMatchInlineSnapshot(`""`);
+      await runWrangler(
+        "publish index.js --compatibility-date 2022-03-17 --name test-name"
+      );
+      expect(std.out).toMatchInlineSnapshot(`
+        "Uploaded test-name (TIMINGS)
+        Published test-name (TIMINGS)
+          test-name.test-sub-domain.workers.dev"
+      `);
+      expect(std.err).toMatchInlineSnapshot(`""`);
+      expect(std.warn).toMatchInlineSnapshot(`""`);
+    });
+  });
+
+  describe("tsconfig", () => {
+    it("should use compilerOptions.paths to resolve modules", async () => {
+      writeWranglerToml({
+        main: "index.ts",
+      });
+      fs.writeFileSync(
+        "index.ts",
+        `import { foo } from '~lib/foo'; export default { fetch() { return new Response(foo)} }`
+      );
+      fs.mkdirSync("lib", { recursive: true });
+      fs.writeFileSync("lib/foo.ts", `export const foo = 123;`);
+      fs.writeFileSync(
+        "tsconfig.json",
+        JSON.stringify({
+          compilerOptions: {
+            baseUrl: ".",
+            paths: {
+              "~lib/*": ["lib/*"],
+            },
+          },
+        })
+      );
+      mockSubDomainRequest();
+      mockUploadWorkerRequest({
+        expectedEntry: "var foo = 123;", // make sure it imported the module correctly
+      });
+      await runWrangler("publish index.ts");
+      expect(std).toMatchInlineSnapshot(`
+        Object {
+          "err": "",
+          "out": "Uploaded test-name (TIMINGS)
+        Published test-name (TIMINGS)
+          test-name.test-sub-domain.workers.dev",
+          "warn": "",
+        }
+      `);
+    });
+
+    it("should output to target es2020 even if tsconfig says otherwise", async () => {
```

Comment:
Should we display a warning in this case?

It could be confusing to developers if that have specified ES5 but it keeps outputting ES2020... They might think that we are not reading tsconfig.json at all?

Reply:
They never see the output so it maaay be ok.